### PR TITLE
[server, installer] Make admin-login-secret configurable (new default: not set)

### PR DESCRIPTION
**Pull request by geropl**
## Description

Depends on: https://github.com/gitpod-io/ops/pull/8127

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by geropl** [comment]
Preparing the preview env for testing...

File: components/server/src/user/user-controller.ts [hunk 1]
```diff
@@ -141,6 +141,11 @@ export class UserController {
         router.get(
             "/login/ots/admin-user/:key",
             loginUserWithOts(async (req: express.Request, res: express.Response, user: User, secret: string) => {
+                // In case there is no/an empty key specified: Nobody should be able to call this so they are not able to guess values here
+                if (!this.config.admin.loginKey) {
+                    throw new ResponseError(500, "Cannot handle request");
```
Comment (easyCZ) [inline components/server/src/user/user-controller.ts:146]: Can we make this nicer, like 
"No Admin login key configured, cannot login as superadmin"

File: install/installer/pkg/components/server/deployment.go [hunk 2]
```diff
@@ -330,19 +330,21 @@ func deployment(ctx *common.RenderContext) ([]runtime.Object, error) {
...
-		MountPath: AdminSecretMountPath,
-		ReadOnly:  true,
-	})
+		})
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      "admin-login-key",
+			MountPath: AdminSecretMountPath,
+			ReadOnly:  true,
+		})
+	}
```
Comment (easyCZ) [inline install/installer/pkg/components/server/deployment.go:347]: Is that key ever optional?

May make it easier if use the `Optional: true` property on the volume-mount, that way you can avoid the extra conditional on the config render

File: components/server/src/installation-admin/installation-admin-controller.ts [hunk 3]
```diff
@@ -50,10 +50,15 @@ export class InstallationAdminController {
...
+                // In case there is no/an empty key specified: Nobody should be able to call this so they are not able to guess values here
+                if (!this.config.admin.loginKey) {
+                    throw new Error("Cannot handle request");
+                }
+
                 // Unblock the admin-user: it's blocked initially!
                 const user = await this.userDb.findUserById(BUILTIN_INSTLLATION_ADMIN_USER_ID);
                 if (!user) {
-                    throw new Error("Cannot find builint admin-user");
+                    throw new Error("Cannot find builtin admin-user");
```
Comment (easyCZ) [inline components/server/src/installation-admin/installation-admin-controller.ts:61]: Use  `BUILTIN_INSTLLATION_ADMIN_USER_ID` to reference the user const

**Review by easyCZ — APPROVED** [review]
/hold 
for comments, your choice on how to address

File: components/server/src/user/user-controller.ts (hunk 1 above)
Comment (geropl) [reply components/server/src/user/user-controller.ts:146]: Done.

File: install/installer/pkg/components/server/deployment.go (hunk 2 above)
Comment (geropl) [reply install/installer/pkg/components/server/deployment.go:347]: Thx for sharing! TIL about that Optional flag. But so far I like the failure mode better (server pod not coming up, vs single error log line). If we want to change that later, we can always revise. :lotus_position:

File: components/server/src/installation-admin/installation-admin-controller.ts (hunk 3 above)
Comment (geropl) [reply components/server/src/installation-admin/installation-admin-controller.ts:61]: I feel the ID is too technical, and referenced the user by it's name throughout user-facing strings. :+1:

**Comment by geropl** [comment]
Manually tested preview env setup and it worked :heavy_check_mark:

**Comment by geropl** [comment]
/hold 

to be able to resolve upcoming installer conflicts with https://github.com/gitpod-io/gitpod/pull/16173

**Review by corneliusludmann — APPROVED** [review]

**Comment by geropl** [comment]
/unhold

Done with rebase

**Pull request merged**